utilities/backup/backup_engine.cc

```diff
@@ -2356,16 +2349,14 @@ IOStatus BackupEngineImpl::ReadFileAndComputeChecksum(
   std::unique_ptr<SequentialFileReader> src_reader;
   auto file_options = FileOptions(src_env_options);
   file_options.temperature = src_temperature;
-  IOStatus io_s = SequentialFileReader::Create(src_fs, src, file_options,
-                                               &src_reader, nullptr);
+  RateLimiter* rate_limiter = options_.backup_rate_limiter.get();
+  IOStatus io_s = SequentialFileReader::Create(
```

Comment:
Not relevant to @cbi42 's impl: for our conversation whether RandomAccessFileReader is used in BackupEngine,  [sst_reader](https://github.com/facebook/rocksdb/blob/0d8433c5853e468975382b05b92cdd72379c64c8/utilities/backup/backup_engine.cc#L2391) [does wrap it and use it .](https://github.com/facebook/rocksdb/blob/main/table/sst_file_dumper.h#L90). 

Though this is not relevant to this PR as it only migrates SequentialFileReader part

Reply:
Should we have a task to track improving `RandomAccessFileReader` with file size?

Reply:
See the follow-up task :p